django/db/backends/mysql/features.py

```diff
@@ -70,6 +69,12 @@ def can_introspect_foreign_keys(self):
         "Confirm support for introspected foreign keys"
         return self._mysql_storage_engine != 'MyISAM'
 
+    @cached_property
+    def introspected_field_types(self):
+        values = super().introspected_field_types.copy()
+        values.update({'BooleanField': 'IntegerField'})
+        return values
```

Comment:
We can use dict unpacking, e.g.
```python
return {
    **super().introspected_field_types,
    'BooleanField': 'IntegerField',
}
```

Reply:
I've just deleted my comments that made the same point as we posted at the same time.

In addition, if removing the defaults in the base, we don't need to use `super()` anymore as the dict is empty and thus we don't need `@cached_property`, or indeed even a `@property`. Just simplify to `introspected_field_types = {'BooleanField': 'IntegerField'}`.

Reply:
I would leave it, we cannot guarantee that it will always be an empty dict.

Also, we could remove some `can_introspect_*` flags and use it instead, e.g.
`can_introspect_binary_field` and base the default `introspected_field_types` on other `can_introspect_*` worth keeping flags, e.g. `can_introspect_autofield`.

Reply:
@felixxm , Should I remove `can_introspect_binary_field` as well :thinking: ?

Reply:
`can_introspect_binary_field` is used only in a single test, so yes. We should audit other `can_introspect_*` flags.

Reply:
Ok, I will check them and include them if possible. Thanks

Reply:
I removed
- can_introspect_binary_field
- can_introspect_decimal_field
- can_introspect_duration_field
- can_introspect_ip_address_field
- can_introspect_time_field

Reply:
It looks to me like can_introspect_big_integer_field, can_introspect_positive_integer_field, and can_introspect_small_integer_field could be removed also. Did you run into an issue?

Reply:
I Removed them as well.